sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -2335,126 +2270,40 @@ public void setup() throws IOException {
         }
       }
 
-      @StartBundle
-      public void startBundle(StartBundleContext context) {
-        batch = ArrayListMultimap.create();
-        currentBatchSizeBytes = 0;
-      }
-
-      /**
-       * Adapter interface which provides a common parent for {@link ProcessContext} and {@link
-       * FinishBundleContext} so that we are able to use a single common invocation to output from.
-       */
-      interface ContextAdapter {
-        void output(
-            TupleTag<Document> tag, Document document, Instant timestamp, BoundedWindow window);
-      }
-
-      private static final class ProcessContextAdapter<T> implements ContextAdapter {
-        private final DoFn<T, Document>.ProcessContext context;
-
-        private ProcessContextAdapter(DoFn<T, Document>.ProcessContext context) {
-          this.context = context;
-        }
-
-        @Override
-        public void output(
-            TupleTag<Document> tag, Document document, Instant ignored1, BoundedWindow ignored2) {
-          // Note: window and timestamp are intentionally unused, but required as params to fit the
-          // interface
-          context.output(tag, document);
-        }
-      }
-
-      private static final class FinishBundleContextAdapter<T> implements ContextAdapter {
-        private final DoFn<T, Document>.FinishBundleContext context;
-
-        private FinishBundleContextAdapter(DoFn<T, Document>.FinishBundleContext context) {
-          this.context = context;
-        }
-
-        @Override
-        public void output(
-            TupleTag<Document> tag, Document document, Instant timestamp, BoundedWindow window) {
-          context.output(tag, document, timestamp, window);
+      @ProcessElement
+      public void processElement(ProcessContext c) throws IOException, InterruptedException {
```

Comment:
Can we avoid `ProcessContext` and use `@Element` and `OutputCollector` for new DoFns? Might be worth changing for the old ones as well, but at least for the new code, we might want to use the new style.